src/main/java/org/opensearch/plugin/insights/core/listener/QueryInsightsListener.java

```diff
@@ -93,11 +114,29 @@ public QueryInsightsListener(
         final ClusterService clusterService,
         final QueryInsightsService queryInsightsService,
         boolean initiallyEnabled
+    ) {
```

Comment:
I believe this constructor should only be used for tests since it is setting `threadpool = null`. Can we consider removing it to reduce the risk of someone using it for production in the future?